templates/display_product.tt.html

```diff
@@ -0,0 +1,243 @@
+            <div class="button_div" id="saveingredientsbuttondiv" style="display:none"><button id="saveingredients" class="small button" type="button">Save ingredients ([% ilc %])</div>
+            
+            <div class="button_div" id="wipeingredientsbuttondiv"><button id="wipeingredients" class="small button" type="button">Ingredients ([% ilc %])) are completely bogus, erase them.</button></div>
+        [% END %]
+        [% display_field_allergens %]
+        [% display_field_traces %]
+        [% display_ingredients_analysis %]
+        [% display_ingredients_analysis_details %]
+        [% IF html_ingredients_classes_n > 0 %]
+            <div class="row">
+                [% html_ingredients_classes %]
+            </div>
+        [% END %]
+
+        <!-- special ingredients tags -->
+        [% IF special_ingredients_tags == 'defined' %]
+            [% IF special_html != "" %]
+                <br><hr class="floatleft">
+                <div>
+                    <b>[% lang("ingredients_analysis") FILTER ucfirst %]:</b><br>
+                    <ul id="special_ingredients">[% special_html %]</ul>
+                    <p>[% lang("ingredients_analysis_note") %]</p>
+                </div>
+                
+            [% END %]
+        [% END %]
+
+        <!-- NOVA Groups -->
+        [% IF product_nova_group == 'exists' %]
+            <h4>[% lang("nova_groups_s") %]
+                <a href="/nova" title="[% a_title %]">
+                [% display_icon('info') %]</a>
+            </h4>
+            <a href="/nova" title="[% a_title %]"><img src="/images/misc/nova-group-[% group %].svg" alt="[% display %]" style="margin-bottom:1rem;max-width:100%"></a><br>
+            [% display %]
+        [% END %]
+    </div>
+    <div class="show-for-large-up large-4 xlarge-4 xxlarge-4 columns" style="padding-left:0">[% html_image %]</div>
+</div>
+
+<!-- nutrition table -->
+[% IF nutrition_table == 'defined' %]
+    <h2>[% lang("nutrition_data") %]</h2>
+    <div class="row">
+        <div class="hide-for-large-up medium-12 columns">
+            [% nutrition_image %]
+        </div>
+        <div class="medium-12 large-8 xlarge-8 xxlarge-8 columns">
+            [% display_nutrient_levels %]
+            [% display_field %]
+            
+            [% IF no_nutrition_data == 'on' %]
+                <div class='panel callout'>
+                    [% lang("no_nutrition_data") %]
+                </div>
+            [% END %]
+            [% display_nutrition_table %]
+        </div>
+    
+        <div class="show-for-large-up large-4 xlarge-4 xxlarge-4 columns" style="padding-left:0">
+            $html_image
+        </div>
+    </div>
+
+    [% IF has_tag == 'categories-en:alcoholic-beverages' %]
+        <p class="panel callout">
+            [% lang('alcohol_warning') %]
+        </p>
+    [% END %]
+[% END %]
+
+<!-- other fields -->
+[% IF other_fields != "" %]
```

Comment:
https://github.com/openfoodfacts/openfoodfacts-server/blob/65b6c9cc6d0e6c080c0e05b909487d278c73f8f5/lib/ProductOpener/Display.pm#L7920-L7922

I think this is only set if it's not empty. How does TT handle the case of the property probably being `undef`?

Reply:
> I think this is only set if it's not empty. How does TT handle the case of the property probably being `undef`?

'undef' fields won't be displayed

Reply:
OK, cool. I was wondering if that might cause any kind of error, but in that case, TT is apparently clever enough. 🙂 Thanks!